projects/ADuCM3029_demo_cn0397/include/cn0397_app.h

```diff
@@ -0,0 +1,108 @@
+  - Redistributions in binary form must reproduce the above copyright notice,
+    this list of conditions and the following disclaimer in the documentation
+    and/or other materials provided with the distribution.
+  - Modified versions of the software must be conspicuously marked as such.
+  - This software is licensed solely and exclusively for use with processors
+    manufactured by or for Analog Devices, Inc.
+  - This software may not be combined or merged with other code in any manner
+    that would cause the software to become subject to terms and conditions
+    which differ from those listed here.
+  - Neither the name of Analog Devices, Inc. nor the names of its
+    contributors may be used to endorse or promote products derived
+    from this software without specific prior written permission.
+  - The use of this software may or may not infringe the patent rights of one
+    or more patent holders.  This license does not release you from the
+    requirement that you obtain separate licenses from these patent holders
+    to use this software.
+
+THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES, INC. AND CONTRIBUTORS "AS IS" AND ANY
+EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+TITLE, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN
+NO EVENT SHALL ANALOG DEVICES, INC. OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, PUNITIVE OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, DAMAGES ARISING OUT OF CLAIMS OF INTELLECTUAL
+PROPERTY RIGHTS INFRINGEMENT; PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+*****************************************************************************/
+#ifndef CN0397_APP_H
+#define CN0397_APP_H
+
+#include <base_sensor/adi_sensor.h>
+#include <visible_light/cn0397/adi_cn0397.h>
+#include <base_sensor/adi_sensor_errors.h>
+#include <adi_ble_config.h>
+#include <drivers/pwr/adi_pwr.h>
+#include <radio/adi_ble_radio.h>
+#include <common/adi_error_handling.h>
+#include <framework/noos/adi_ble_noos.h>
+#include <common/adi_timestamp.h>
+
+/* Visible light instance ID */
+#define ADI_VISIBLE_LIGHT_ID    (1u)
+
+/* BLE event processing dispatcher timeout. Waits for 2 seconds for BLE events */
+#define ADI_APP_DISPATCH_TIMEOUT (2000)
+
+/* To use Bluetooth for this test, set this to 1. To use the console for output set this to 0. */
+#define ADI_APP_USE_BLUETOOTH (1u)
+
+/* Defined in pinmux.c */
+extern "C" int32_t adi_initpinmux(void);
+
+typedef union {
+	uint8_t u8Value[4];
+	float fValue;
+} floatUnion_t;
+
+#pragma pack(push)
+#pragma pack(1)
+typedef struct GenericReg {
```

Comment:
We should avoid using "typedef" unless creating opaque structure pointers meant only to be accessed by controlled API or abstracting too complex data types like function pointers.